src/cached_storage.rs

```diff
@@ -251,6 +251,16 @@ impl<S: StorageExt> StorageExt for CachedStorage<S> {
         self.storage.append(entries)
     }
 
+    // The cache resides in memory and is not persistent, so we only care about setting/getting the
+    // last applied entry in the underlying storage (which may be persistent)
+    fn applied(&self) -> Result<u64, raft::Error> {
```

Comment:
I am fine with omitting this from the cache, but my reasoning is that you only read this once on startup. I'm not sure the reasoning in the comment makes sense.